### DISCUSSION TITLE
Apple HEIC files not converted

### DISCUSSION THREAD

**Discussion by newwaylw** [photoprism/photoprism#3592] [Q&A] [answered]
Hi all, I've been using the latest photoprism release on raspberry pi 4 using docker compose:
> services:
> photoprism:
> image: photoprism/photoprism:arm64

When indexing my original photo files that contained Apple HEIC image files, i noticed that these files are not converted to jpg (therefore not displayed in photoprism).
I executed into the docker container and confirmed that:

- no HEIC.jpg file are created in sidecar/ folder 
- when manually execute `heif-convert xxx.HEIC xxx.jpg` in the container I get error message `Input file is not an HEIF/AVIF file`
- `heif-convert --help` shows `heif-convert  libheif version: 1.13.0`

>   root@230719:/photoprism$ heif-convert --help
> heif-convert  libheif version: 1.13.0
>         -------------------------------------------
> Usage: heif-convert [options]  <input-image> <output-image>
>
>The program determines the output file format from the output filename suffix.
>These suffices are recognized: jpg, jpeg, png, y4m.
>Options:
 > -h, --help      show help
 > -q, --quality   quality (for JPEG output)
 >     --with-aux  also write auxiliary images (e.g. depth images)
 >     --no-colons replace ':' characters in auxiliary image filenames with '_'
 >     --quiet     do not output status messages to console

>root@230719:/photoprism$ /usr/local/bin/heif-convert IMG_6710.HEIC test.jpg
>Input file is not an HEIF/AVIF file

I've seem similar issues reported last year but it seems this is resolved with a newer image, any pointers is much appreciated.
Thank you

**Comment by lastzero** [comment, marked as answer]
These are probably JPEG files then.

**Reply by newwaylw** [reply]
Thanks for the reply,  `file` does say it is a JPEG:

> root@230719:/photoprism$ file IMG_6710.HEIC
> IMG_6710.HEIC: JPEG image data, JFIF standard 1.01, aspect ratio, density 1x1, segment length 16, Exif Standard: [TIFF image data, > big-endian, direntries=12, manufacturer=Apple, model=iPhone 12, xresolution=174, yresolution=182, resolutionunit=2, software=15.5, datetime=2022:07:26 23:05:03, hostcomputer=iPhone 12], baseline, precision 8, 4030x2267, components 3

Then here comes the next question, why are all these files not shown in photoprism?

**Reply by lastzero** [reply]
Because the extension belongs to the HEIC container format and not JPEG.

**Reply by newwaylw** [reply]
Thank you.
but I am confused,  https://www.photoprism.app/kb/file-formats mentioned it supports HEIC

**Reply by lastzero** [reply]
They are JPEG images, not HEIC. Problem is they use the wrong file extension.

**Reply by graciousgrey** [reply]
If you correct the wrong file extension and start indexing again, the images should appear as expected.

**Comment by steven-ellis** [comment]
This is a major issue if you're using a tool like rclone to backup files from Google Photos. The synced files end with HEIC, but in reality are jpegs. Is there any way to to update the scanner to check the files in case they are JPEGs?
